packages/jupyterlab-manager/src/manager.ts

```diff
@@ -504,10 +505,14 @@ export class KernelWidgetManager extends LabWidgetManager {
    * Restore widgets from kernel and saved state.
    */
   async restoreWidgets(): Promise<void> {
-    await this._loadFromKernel();
-    this._restoredStatus = true;
-    this._initialRestoredStatus = true;
-    this._restored.emit();
+    try {
+      await this._loadFromKernel();
+      this._restoredStatus = true;
+      this._initialRestoredStatus = true;
+      this._restored.emit();
+    } catch (err) {
+      // Do nothing
```

Comment:
This shouldn't really happen for the `KernelWidgetManager`, but I agree with being defensive.

Reply:
Do you mean there won't be an error in loading from kernel in the widget manager? We aren't throwing one explicitly, but of course there could be one, and that would leave us in a weird state for the flag, so I think it's important to be defensive.

Probably even better semantically to have it in a finally statement, though.

Reply:
Or maybe even better to move the flag setting into _loadFromKernel...